src/EntityManager.php

```diff
@@ -198,6 +198,27 @@ public function getList(){
 		return \$this->findAll();
 	}
 	
+	/**
+     * Finds only one entity. The criteria must contain all the elements needed to find a unique entity.
+     * Throw an exception if more than one entity was found.
+     *
+     * @param array \$criteria
+     *
+     * @return ".$entityName." the bean object
+     */
+    public function findTheOnlyOneBy(array \$criteria)
+    {
+        \$result = \$this->findBy(\$criteria);
+
+        if(count(\$result) == 1){
+            return \$result[0];
+        }elseif(count(\$result) > 1){
+            throw new NonUniqueResultException;
```

Comment:
I don't see the NonUniqueResultExcetpion.php file.
Also, you should put a message as parameter, passing as much data as possible (in particular, the message should contain the name of the entitity retrieved and possibly the list of criteria).